src/SqlPersistence/SynchronizedStorage/SqlStorageSessionFeature.cs

```diff
@@ -12,13 +12,9 @@ protected override void Setup(FeatureConfigurationContext context)
         var services = context.Services;
         var connectionManager = settings.GetConnectionBuilder<StorageType.Sagas>();
 
-        var sessionHolder = new CurrentSessionHolder();
-
         //Info cache can be null if Outbox is enabled but Sagas are disabled.
-        services.AddSingleton<ISynchronizedStorage>(provider => new SynchronizedStorage(connectionManager, provider.GetService<SagaInfoCache>(), sessionHolder));
-        services.AddSingleton<ISynchronizedStorageAdapter>(provider => new StorageAdapter(connectionManager, provider.GetService<SagaInfoCache>(), sqlDialect, sessionHolder));
-
-        services.AddTransient(_ => sessionHolder.Current);
-        context.Pipeline.Register(new CurrentSessionBehavior(sessionHolder), "Manages the lifecycle of the current session holder.");
+        // TODO: The connection manager design needs a revisit to avoid the closure
```

Comment:
Is this done or what? :grumpy-face:

Reply:
See https://github.com/Particular/NServiceBus.Persistence.Sql/pull/915/commits/79411ffb2679789abdfab7b07a11a912725e0801